compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -271,7 +271,7 @@ impl<'tcx> TypeVisitor<TyCtxt<'tcx>> for ImplTraitInTraitFinder<'_, 'tcx> {
 
     fn visit_ty(&mut self, ty: Ty<'tcx>) -> std::ops::ControlFlow<Self::BreakTy> {
         if let ty::Alias(ty::Projection, alias_ty) = *ty.kind()
-            && self.tcx.def_kind(alias_ty.def_id) == DefKind::ImplTraitPlaceholder
+            && self.tcx.is_impl_trait_in_trait(alias_ty.def_id)
             && self.tcx.impl_trait_in_trait_parent_fn(alias_ty.def_id) == self.fn_def_id
             && self.seen.insert(alias_ty.def_id)
```

Comment:
The logic on 291/292 is not correct now. Could you fix it, or at least add a `FIXME(-Z.....): ...` with some sort of explanation like "need to project to the opaque, could get it via type_of + subst"

Reply:
I'm not sure I quite follow what's wrong here. Was looking at the comments in https://github.com/rust-lang/rust/pull/109140/commits/81ac6e69454868a78d8074892d1ab4a3445d118f#diff-2c3dca0b66724d6aadfabd5335e0441089497ee6ef91f960bdcd5d86b36101d4L248-L251 and what we are already doing is installing a projection -> opaque predicate into the predicate list. Isn't just doing that enough? or are we installing the predicate incorrectly for the new RPITITs?.

Reply:
That def-id is not an opaque type with the new lowering strategy.

Reply:
Ohh yeah, right, this type is a projection :+1:. I guess we can leave with a FIXME on this PR and fix it afterward.